crates/node/src/catchup.rs

```diff
@@ -65,28 +65,29 @@ impl Node {
 
         let application_id = context.application_id.clone();
 
-        let Some(url) = request.url else {
+        let Some(url) = request.source else {
             eyre::bail!("Path is missing in the request")
         };
 
-        let Some(hash) = request.hash else {
-            eyre::bail!("Hash is missing in the request")
-        };
-
         if request
             .application_id
             .map_or(true, |id| id != application_id)
         {
-            let application_version = self
-                .ctx_manager
-                .get_application_latest_version(&application_id)?;
+            let Some(application) = self.ctx_manager.get_application(&application_id)? else {
+                eyre::bail!(
+                    "fatal error: context `{}` links to dangling application ID `{}`",
+                    context.id,
+                    application_id
+                );
+            };
 
             let message = serde_json::to_vec(&types::CatchupStreamMessage::ApplicationChanged(
                 types::CatchupApplicationChanged {
                     application_id,
-                    version: application_version,
-                    url,
-                    hash,
+                    blob_id: application.blob,
+                    version: application.version,
+                    source: url,
+                    hash: None, // todo! blob_mgr(application.blob)?.hash
```

Comment:
highlighting for visibility

we should be able to share the expected hash here once blob mgr returns a hash